test/unit/test_stepsize.py

```diff
@@ -38,6 +38,20 @@ def test_hit_stepsize_max(self, my_stepsize):
         new_value = float(my_stepsize.value)
         assert new_value == my_stepsize.adaptive_stepsize["stepsize_stop_max"]
 
+    def test_hit_stepsize_max_float(self, my_stepsize):
+        my_stepsize.value.assign(10)
+        my_stepsize.adaptive_stepsize["max_stepsize"] = 1
+        my_stepsize.adapt(t=6, converged=True, nb_it=2)
+        new_value = float(my_stepsize.value)
+        assert new_value == my_stepsize.adaptive_stepsize["max_stepsize"]
+
+    def test_hit_stepsize_max_callable(self, my_stepsize):
+        my_stepsize.value.assign(10)
+        my_stepsize.adaptive_stepsize["max_stepsize"] = lambda t: t * 0 + 1
```

Comment:
Do you need t*0 here? Would this work?
```suggestion
        my_stepsize.adaptive_stepsize["max_stepsize"] = lambda t: 1
```

Reply:
I updated the test